my-code/src/Contexts/MoviesData.tsx

```diff
@@ -0,0 +1,95 @@
+import axios from "axios";
+import { createContext, ReactNode, useContext, useEffect, useState } from "react";
+
+interface MoviesDataProviderProps {
+    children: ReactNode;
+}
+
+interface MoviesDataContextProps {
+    movies: MovieProps[];
+    getMoviesData: (search: string) => Promise<void>;
+    favouriteMovies: FavouriteMoviesProps[];
+    addToFavourites: (movieID: string) => void;
+    isFavourite: (movieID: string) => boolean;
+    isLoading: boolean;
+}
+
+interface MovieProps {
+    Actors: string;
+    Awards: string;
+    Country: string;
+    Director: string;
+    Genre: string;
+    Language: string;
+    Metascore: string;
+    Plot: string;
+    Poster: string;
+    Released: string;
+    Title: string;
+    Year: string;
+    imdbRating: string;
+    imdbID: string;
+}
+
+interface FavouriteMoviesProps {
+    imdbID: string;
+}
+
+const MoviesDataContext = createContext({} as MoviesDataContextProps)
+
+export function MoviesDataProvider({ children }: MoviesDataProviderProps) {
+
+    const [movies, setMovies] = useState<MovieProps[]>([])
+    const [favouriteMovies, setFavouriteMovies] = useState<FavouriteMoviesProps[]>(
+        JSON.parse(localStorage.getItem('favouriteMovies') || '[]')
+    )
+    const [isLoading, setIsLoading] = useState(false)
+
+    async function getMoviesData(search: string) {
+        if (search.length > 0 && search.length < 3) return;
+
+        setIsLoading(true)
+        if (search.length === 0) return (
+            setMovies([]),
+            setIsLoading(false)
+        )

```

Comment:
You have a line already that checks if search.length is less than zero. Shouldn't you include this check there? Unless you always want to do setIsLoading(false) with invalid search lengths... In that case:
```suggestion
        if (search.length >= 0) {
            setMovies([]),
            setIsLoading(false)
            return
        }

        setIsLoading(true)
```

Reply:
Line 49 was to "limit" the requests, but i learned today about debounce (thank you) so, i agree that it would be useless.
Your suggestion on line 50~ wouldnt work, since:
````
if (search.length >= 0) {
            setMovies([]),
````
Would always reset the movies array each time the user searches for something.

I guess this would be a better solution:
````
if (search.length === 0) {
            setMovies([]),
            setIsLoading(false)
            return
        }
````

Reply:
Yes, that was a typo, sorry! Your solution is better 🙂